src/wp-includes/wp-db.php

```diff
@@ -2018,12 +2018,12 @@ public function query( $query ) {
 				$this->insert_id  = 0;
 				$this->last_query = $query;
 
-				if ( function_exists( '__' ) ) {
-					$this->last_error = __( 'WordPress database error: Could not perform query because it contains invalid data.' );
-				} else {
-					$this->last_error = 'WordPress database error: Could not perform query because it contains invalid data.';
+				if ( ! function_exists( '__' ) ) {
+					wp_load_translations_early();
```

Comment:
@SergeyBiryukov What do you think? I used the same approach for `wpdb::query()` and `wpdb::process_fields()` where if the function isn't loaded into memory yet and the error message is going to be set, then invoke `wp_load_translations_early()`.

Reply:
I think the conditional is redundant, as `wp_load_translations_early()` is called without that check in other methods, so I would prefer the same here for consistency. The function has an early exit if translations are already loaded, I don't see a reason to duplicate the check here. Looks good to me othewise.

Reply:
@SergeyBiryukov removed `functions_exists()` in commit https://github.com/WordPress/wordpress-develop/pull/1897/commits/6f12dad2b7772316d6636af606d5af912cad3ade.